spec/requests/api/internal/v2/court_application_laa_references_spec.rb

```diff
@@ -161,35 +166,38 @@
           let(:Authorization) { "Bearer #{token.token}" }
 
           before do
-            expect(UnlinkCourtApplicationLaaReferenceWorker).to receive(:perform_async).with("XYZ", subject_id, "JaneDoe", 1, "", 1_231_231)
+            expect(CourtApplicationLaaReferenceUnlinker).to receive(:call).with(subject_id:,
+                                                                                user_name: "JaneDoe",
+                                                                                unlink_reason_code: 1,
+                                                                                unlink_other_reason_text: "",
+                                                                                maat_reference: 1_231_231)
           end
 
           run_test!
         end
 
         context "when data is bad" do
-          response("422", "Unprocessable Entity") do
-            let(:Authorization) { "Bearer #{token.token}" }
-            let(:subject_id) { "X" }
+          let(:Authorization) { "Bearer #{token.token}" }
+          let(:subject_id) { "X" }
 
+          response("422", "Unprocessable Entity") do
             before do
-              expect(UnlinkLaaReferenceWorker).not_to receive(:perform_async)
+              expect(CourtApplicationLaaReferenceUnlinker).not_to receive(:call)
             end
 
             run_test!
           end
         end
 
         context "when subject_id is nonexistent" do
-          response("404", "Defendant not found") do
-            let(:Authorization) { "Bearer #{token.token}" }
-
-            let(:subject_id) { "fa7ca7bd-5dce-419c-88db-f42e1b7ce8a0" }
+          let(:Authorization) { "Bearer #{token.token}" }
+          let(:subject_id) { "fa7ca7bd-5dce-419c-88db-f42e1b7ce8a0" }
 
+          response("404", "Defendant not found") do
             run_test! do |response|
               error = JSON.parse(response.body)["error"]
 
-              expect(error).to include("Couldn't find LaaReference")
+              expect(error).to include("Defendant not found!")
```

Comment:
When the call to Common Platform fails, what status code do you want to be returned? You should add a spec demonstrating that to this file. (Hint, for Prosecution Case LAA References we return a 424, and you can see the spec at https://github.com/ministryofjustice/laa-court-data-adaptor/blob/main/spec/requests/api/internal/v2/laa_references_spec.rb#L65

Reply:
I've fixed it this way: any possible connection error with CP, will be converted to 424 - Failed Dependency.

Reply:
I'd be wary of trying to list every possible Faraday error in the ApplicationController. For one thing as currently configured, if CP returns a 404 status code that _won't_ raise a Faraday::ResourceNotFound, and the service layer will raise an unhandled StandardError. For another, Faraday errors don't have `error_code` attributes, so won't be handled very nicely, whereas our own CommonPlatform::Errors::FailedDependency _does_ have an error code. So ideally you'd catch any errors lower down and wrap them up into our own error class.

Reply:
ok, it makes sense,  I'll chage it to raise the`CommonPlatform::Api::Errors::FailedDependency` 👍